app/controllers/spree/admin/images_controller_decorator.rb

```diff
@@ -0,0 +1,6 @@
+Spree::Admin::ImagesController.class_eval do
+  # This will make resource controller redirect correctly after deleting product images.
+  # This can be removed after upgrading to Spree 2.1.
+  # See here https://github.com/spree/spree/commit/334a011d2b8e16355e4ae77ae07cd93f7cbc8fd1
+  belongs_to 'spree/product', :find_by => :permalink
```

Comment:
sorry, I don't understand what this line has to do with that commit. I don't even know what a `belongs_to` does in a controller. What did you find out?

Reply:
it's how resource controller works. this is how the collection_url is build:
https://github.com/openfoodfoundation/spree/blob/2-0-stable/backend/app/controllers/spree/admin/resource_controller.rb#L242

belongs_to is used to define the parent of the resource (in terms of routes) and then collection_url will use the parent to build the path, in this case, instead of resulting in admin_images_path (it doesnt exist) it builds the path correctly by using parent admin_product_images_path. this is how I fix it here.
The fix in spree is different, somewhere in the future they create a resource_controller. location_after_destroy that image_controller can override. I think myy fix is easier, and, anyway, can be deleted when upgrading to 2.1.